frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -104,6 +108,12 @@ public final class EnhancedGraphTable extends BaseEnhancedGraphTable
   @UIEffect
   private @Nullable String selectedBranchName;
 
+  // To accurately track the change of the current branch from the begging, let's put something impossible
+  // as a branch name. This is required to detect the moment when the unmanaged branch notification should be shown.
+  private @Nullable String currentBranch = "?!@#$%^&";
```

Comment:
```suggestion
  private String currentBranch = "?!@#$%^&";
```

Making it nullable suggests that we also care about detached HEAD state... which AFAICS we don't.
And if we don't care about detached HEAD indeed... then maybe a rename would also make sense:

```suggestion
  private String mostRecentlyCheckedOutBranch = "?!@#$%^&";
```